From ab1c8f2a914300783f769819bed37c848c66721a Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Mon, 7 Oct 2019 15:25:42 +0300
Subject: [PATCH] Fix segfault at loading older format savegame

Conversion of the action_vector was trying to save excess, uninitialized,
elements.

See hrm Bug #840623

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/savegame/savecompat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/savegame/savecompat.c b/server/savegame/savecompat.c
index 82a3ea33b6..cca439281d 100644
--- a/server/savegame/savecompat.c
+++ b/server/savegame/savecompat.c
@@ -1766,7 +1766,7 @@ static void compat_load_dev(struct loaddata *loading)
         }
       }
 
-      secfile_replace_str_vec(loading->file, modname_new, NUM_ACTIONS,
+      secfile_replace_str_vec(loading->file, modname_new, action_count,
                               "savefile.action_vector");
 
       free(modname_new);
-- 
2.20.1

